sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusSupplementaryAuthHeaderPolicy.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.administration;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.messaging.servicebus.implementation.ServiceBusConstants;
+import com.azure.messaging.servicebus.implementation.ServiceBusSharedKeyCredential;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+
+import static com.azure.messaging.servicebus.implementation.ServiceBusConstants.SERVICE_BUS_DLQ_SUPPLEMENTARY_AUTHORIZATION_HEADER_NAME;
+import static com.azure.messaging.servicebus.implementation.ServiceBusConstants.SERVICE_BUS_SUPPLEMENTARY_AUTHORIZATION_HEADER_NAME;
+
+/**
+ * Authentication policy to add necessary supplementary auth headers when forwarding is set.
+ */
+public class ServiceBusSupplementaryAuthHeaderPolicy implements HttpPipelinePolicy {
```

Comment:
This shouldn't be part of the public surface area, package-private.

Reply:
agreed, was just trying to keep this consistent. 
changed this to package-private for now, will need to change others as well

Reply:
tried setting it to package-private, but stylecheck threw up saying it needs to be public